sql/parser/builtins.go

```diff
@@ -48,9 +48,15 @@ var errRoundNumberDigits = errors.New("number of digits must be greater than 0")
 type typeList []reflect.Type
 
 type builtin struct {
-	// Set to typeList{} for nullary functions and to nil for varidic
-	// functions.
+	// Variadic functions have a tricky representation:
+	// - If all arguments must be of a specific type, `types` must be
+	// `typeList{<theType>}`, and `variadic` must be true. This will
+	// generate the correct type-checking error.
+	// - If all arguments must be of the same (variable) type, `types`
```

Comment:
Ah, this was existing code. You don't seem to use this with `variadic == true` which made this comment slightly confusing as it is now written.